fdbclient/S3BlobStore.actor.cpp

```diff
@@ -310,10 +310,11 @@ Reference<S3BlobStoreEndpoint> S3BlobStoreEndpoint::fromString(const std::string
 			}
 
 			// The parameter is known to S3BlobStoreEndpoint so it must be numeric and valid.
-			char* valueEnd;
-			int ivalue = strtol(value.toString().c_str(), &valueEnd, 10);
-			if (*valueEnd || (ivalue == 0 && value.toString() != "0"))
-				throw format("%s is not a valid value for %s", value.toString().c_str(), name.toString().c_str());
+			char* valueEnd = nullptr;
+			std::string s = value.toString();
+			int ivalue = strtol(s.c_str(), &valueEnd, 10);
+			if (*valueEnd || (ivalue == 0 && s != "0"))
+				throw format("%s is not a valid value for %s", s.c_str(), name.toString().c_str());
```

Comment:
if `*valueEnd` is not zero, then it means two cases
* value's first number is invalid, as there are remaining invalid characters in the string
* value has more than 1 valid numbers

It seems we only want to consider it an error in the first case, is the second case possible?

BTW, i think `valueEnd` will be changed by `strtol` , to point to either
* the first invalid character after the first valid number
* "\0" if there is only 1 number in the string and it is valid.
So it seems that it does not matter if we initialize it with a value.. I think?

Reply:
This is parsing URL encoded parameters. So the value should only have 1 valid number.

`valueEnd` initialized to `nullptr` does not have any effect in this case, but it's a good practice to set an initial value.

Reply:
This PR fixes the use-after-free memory issue for `valueEnd`, and doesn't change the original code logic.

Reply:
makes sense, thanks